pxr/usdValidation/usdUtilsValidators/validators.cpp

```diff
@@ -97,12 +98,55 @@ _PackageEncapsulationValidator(const UsdStagePtr &usdStage)
     return errors;
 }
 
+static UsdValidationErrorVector
+_CompressionValidator(const UsdStagePtr& usdStage) {
```

Comment:
@tallytalwar I've come across something in this validator that's not clear to me.

My expectation from reading the description from the existing [CompressionChecker](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdUtils/complianceChecker.py#L149) is that this test should make sure that files within the USDZ file itself were not compressed. So I've created a sample file like this (in this [repo](https://github.com/beersandrew/usd-test-assets/tree/main/Validation/examples/CompressionChecker) as compressedUsdaInside.usdz). However this file fails [here](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/sdf/textFileFormat.cpp#L161), before it can get to the usdchecker code.

If I try a different approach of compressing the usdz itself, it also fails early [here](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usd/usdzResolver.cpp#L217).

Do you know how to reproduce the nicer errors from usdchecker itself? (Similar situation with ByteAlignmentChecker as well)

Reply:
1. Why its failing on parsing the "compressed" "usda" file is because I think usda extension makes it load the usda file plugin, which when parsing tries to see if the file header has the usda magic cookie and if not it fails. If you stick with this approach only maybe use a different extension for this file.
2. usdz is an archival format, so the code checks if it has any compression on it or not.

Maybe add a compressed asset to usdz file like a compressed jpeg/png?

Reply:
I've tried compressing a jpg and adding it to the usdz, and I've tried including a zip file as an asset and then including it in the usdz as well. I'm still unable to trigger the CompressionChecker in the existing `complianceChecker.py` usdchecker version